### ISSUE TITLE
Arrow functions and iife

### ISSUE THREAD

**Issue by salzhrani** [eslint/eslint#4633]
eg: 
`var foo = ()=>{return 'bar'}();`
raises an error
`1:14 - Parsing error: Unexpected token = (undefined)`
and 
`var foo = function(){return 'bar'}();`
passes

version `v1.10.3`
Not sure how to fix

**Comment by eslintbot** [comment]
@salzhrani Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by mysticatea** [comment]
Yes, it's according to the spec: http://www.ecma-international.org/ecma-262/6.0/#sec-expressions

`ArrowFunction` is a `AssignmentExpression`, so it cannot become a callee.
`FunctionExpression` is a `PrimaryExpression`, so it can become a callee.

You need parentheses here.

```js
var foo = (()=>{return 'bar'})();
```

**Comment by salzhrani** [comment]
Got it ...

Thanks
